.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,20 @@
+name: "CodeQL"
+
+on: [pull_request]
+jobs:
+  lint:
+    name: CodeQL
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    - run: git checkout HEAD^2
```

Comment:
And I don't understand why we need to do `HEAD^2` ? what is the significance of 2 here ?

Reply:
Not sure, copied it from somewhere. Replacing with what we have in release.yml, without this twos.